src/namingService.ts

```diff
@@ -0,0 +1,13 @@
+import { SourceDefinition } from './types';
+
+export default abstract class NamingService {
+  abstract normalizeSourceDefinition(
+    source: boolean | string | SourceDefinition,
+  );
+
+  normalizeSource(
```

Comment:
You would really need this function only in case it takes more than 1 line of code: currently it serves no purpose and you can call `normalizeSourceDefinition` from subclasses directly.

Reply:
I think this makes a more readable code (normalizeSource vs normalizeSourceDefinition). and we might still enjoy having abstract interface for future naming services.